spyder/plugins/help/plugin.py

```diff
@@ -140,6 +141,48 @@ def on_main_menu_available(self):
         else:
             self._setup_menus()
 
+    @on_plugin_teardown(plugin=Plugins.Console)
+    def on_console_teardown(self):
+        widget = self.get_widget()
+        internal_console = self.get_plugin(Plugins.Console)
+        internal_console.sig_help_requested.disconnect(self.set_object_text)
+        widget.set_internal_console(None)
+
+    @on_plugin_teardown(plugin=Plugins.Editor)
+    def on_editor_teardown(self):
+        editor = self.get_plugin(Plugins.Editor)
+        editor.sig_help_requested.disconnect(self.set_editor_doc)
+
+    @on_plugin_teardown(plugin=Plugins.IPythonConsole)
+    def on_ipython_console_teardown(self):
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        ipyconsole.sig_shellwidget_changed.disconnect(self.set_shellwidget)
+        ipyconsole.sig_shellwidget_created.disconnect(
+            self.set_shellwidget)
+        ipyconsole.sig_render_plain_text_requested.disconnect(
+            self.show_plain_text)
+        ipyconsole.sig_render_rich_text_requested.disconnect(
+            self.show_rich_text)
+
+        ipyconsole.sig_help_requested.disconnect(self.set_object_text)
+
+    @on_plugin_teardown(plugin=Plugins.Preferences)
+    def on_preferences_teardown(self):
+        preferences = self.get_plugin(Plugins.Preferences)
+        preferences.deregister_plugin_preferences(self)
+
+    @on_plugin_teardown(plugin=Plugins.Shortcuts)
+    def on_shortcuts_teardown(self):
+        shortcuts = self.get_plugin(Plugins.Shortcuts)
+        self.shortcuts_available = False

```

Comment:
This attribute is not being used anywhere else in this plugin. Is it really necessary?